src/main/java/racingcar/domain/Cars.java

```diff
@@ -1,32 +1,34 @@
 package racingcar.domain;
 
-import java.util.ArrayList;
+import java.util.Comparator;
 import java.util.List;
+import java.util.NoSuchElementException;
 import java.util.Objects;
 import java.util.stream.Collectors;
 import racingcar.domain.car.Car;
 
 public class Cars {
 
-    public static final String NEWLINE = "\n";
     private final List<Car> cars;
 
     private Cars(List<Car> carList) {
         this.cars = carList;
     }
 
-    public static Cars createWithDefaults(int numberOfCars) {
-        List<Car> rawCars = new ArrayList<>();
-        for (int i = 0; i < numberOfCars; i++) {
-            rawCars.add(new Car());
-        }
-        return new Cars(rawCars);
+    public static Cars createWithNames(List<String> names) {
+        List<Car> carList = names.stream().map(Car::createWithName)
+                .collect(Collectors.toList());
+        return new Cars(carList);
```

Comment:
```suggestion
        return names.stream().map(Car::createWithName)
                .collect(Collectors.collectingAndThen(Collectors.toList(), Cars::new));
```

와 같이 사용하는 방법도 존재합니다 :)

Reply:
좋습니다. 한줄로 끝낼 때 뭔가 알 수 없는 희열을 느낍니다.